OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -1774,6 +1774,19 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				// Split Locomotors from Mobile
+				if (engineVersion < 20180316)
+				{
+					var mobile = node.Value.Nodes.FirstOrDefault(n => n.Key == "Mobile");
+					if (mobile != null)
+					{
+						// Writing a 'proper' upgrade rule that works well would be extremely hard due to inheritance and World involvement,
+						// notifying modders but leaving the actual changes to them is likely the better approach for everyone.
+						Console.WriteLine("Various properties of Mobile have been split off to Locomotor World trait.");
```

Comment:
Can you please include a #14964 compatible update rule, following the example of `DefineSoundDefaults` to print out more concrete explanations and yaml locations?

Reply:
In the `AfterUpdate` you should first yield return a message explaining how to define locomotor types on the world actor, and then a composite message with all the Mobile references that may need to be looked at.